### ISSUE TITLE
no longer processing after update - message: TypeError

### ISSUE THREAD

**Issue by fugo** [getgrav/grav-plugin-form#591]
Form  v7.1.2
Grav v1.7.39.4

After Update (don't now exactly which one) forms on different websites don't work anymore.
The processing stops with:

```php
TypeError
mb_strpos(): Argument #1 ($haystack) must be of type string, array give
```
The layout is correct, but after sending I get the error.  
I tried also the easy form from [Grav documentation](https://learn.getgrav.org/17/forms/forms) 

I have to downgrade the websites, I think.

**Comment by rhukster** [comment]
Can you turn on full stack traces and let me know exactly what file and what line that error is coming from? Or check in your `logs/grav.log` file?

**Comment by fugo** [comment]
In the logs I don't find a specific error, but after downgrade grav and form i found the error belongs to the email plugin (testing 4.0)
Some had the same [issue](https://github.com/getgrav/grav-plugin-email/issues/167#issue-1482416746) https://github.com/getgrav/grav-plugin-email/issues/167
I think, we can close this

**Comment by fugo** [comment]
the log:

```php
grav.CRITICAL:
 mb_strpos(): Argument #1 ($haystack) must be of type string, array given - 
Trace:   
#0 --/public_html/system/src/Grav/Common/Utils.php(253): mb_strpos()  
#1 --/public_html/user/plugins/email/classes/Email.php(255): Grav\Common\Utils::contains()  
#2 --/public_html/user/plugins/email/classes/Email.php(204): Grav\Plugin\Email\Email->processRecipients()  
#3 --/public_html/user/plugins/email/email.php(127): Grav\Plugin\Email\Email->buildMessage()  
#4 --/public_html/user/plugins/email/email.php(113): Grav\Plugin\EmailPlugin->sendFormEmail()  
#5 --/public_html/vendor/symfony/event-dispatcher/EventDispatcher.php(264): Grav\Plugin\EmailPlugin->onFormProcessed()  
#6 --/public_html/vendor/symfony/event-dispatcher/EventDispatcher.php(239): Symfony\Component\EventDispatcher\EventDispatcher->doDispatch()  
#7 --/public_html/vendor/symfony/event-dispatcher/EventDispatcher.php(73): Symfony\Component\EventDispatcher\EventDispatcher->callListeners()  
#8 --/public_html/system/src/Grav/Common/Grav.php(597): Symfony\Component\EventDispatcher\EventDispatcher->dispatch()  
#9 --/public_html/user/plugins/form/classes/Form.php(940): Grav\Common\Grav->fireEvent()  
#10 --/public_html/user/plugins/form/form.php(279): Grav\Plugin\Form\Form->post()  
#11 --/public_html/vendor/symfony/event-dispatcher/EventDispatcher.php(264): Grav\Plugin\FormPlugin->onPageInitialized()  
#12 --/public_html/vendor/symfony/event-dispatcher/EventDispatcher.php(239): Symfony\Component\EventDispatcher\EventDispatcher->doDispatch()  
#13 --/public_html/vendor/symfony/event-dispatcher/EventDispatcher.php(73): Symfony\Component\EventDispatcher\EventDispatcher->callListeners()  
#14 --/public_html/system/src/Grav/Common/Grav.php(597): Symfony\Component\EventDispatcher\EventDispatcher->dispatch()  
#15 --/public_html/system/src/Grav/Common/Processors/PagesProcessor.php(60): Grav\Common\Grav->fireEvent()  
#16 --/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(50): Grav\Common\Processors\PagesProcessor->process()  
#17 --/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(62): Grav\Framework\RequestHandler\RequestHandler->handle()  
#18 --/public_html/system/src/Grav/Common/Processors/TwigProcessor.php(38): Grav\Framework\RequestHandler\RequestHandler->handle()  
#19 --/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(50): Grav\Common\Processors\TwigProcessor->process()  
#20 --/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(62): Grav\Framework\RequestHandler\RequestHandler->handle()  
#21 --/public_html/system/src/Grav/Common/Processors/AssetsProcessor.php(39): Grav\Framework\RequestHandler\RequestHandler->handle()  
#22 --/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(50): Grav\Common\Processors\AssetsProcessor->process()  #23 --/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(62): Grav\Framework\RequestHandler\RequestHandler->handle()  
#24 --/public_html/system/src/Grav/Common/Processors/SchedulerProcessor.php(40): Grav\Framework\RequestHandler\RequestHandler->handle()  
#25 --/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(50): Grav\Common\Processors\SchedulerProcessor->process()  #26 --/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(62): Grav\Framework\RequestHandler\RequestHandler->handle()  
#27 --/public_html/system/src/Grav/Common/Processors/BackupsProcessor.php(39): Grav\Framework\RequestHandler\RequestHandler->handle()  
#28 --/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(50): Grav\Common\Processors\BackupsProcessor->process()  #29 --/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(62): Grav\Framework\RequestHandler\RequestHandler->handle() 
30 --/public_html/system/src/Grav/Common/Processors/TasksProcessor.php(69): Grav\Framework\RequestHandler\RequestHandler->handle()  
#31 --/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(50): Grav\Common\Processors\TasksProcessor->process()  
#32 --/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(62): Grav\Framework\RequestHandler\RequestHandler->handle()  
#33 --/public_html/system/src/Grav/Common/Processors/RequestProcessor.php(64): Grav\Framework\RequestHandler\RequestHandler->handle()  
#34 --/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(50): Grav\Common\Processors\RequestProcessor->process()  
#35 --/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(62): Grav\Framework\RequestHandler\RequestHandler->handle()  
#36 --/public_html/system/src/Grav/Common/Processors/ThemesProcessor.php(38): Grav\Framework\RequestHandler\RequestHandler->handle()  
#37 --/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(50): Grav\Common\Processors\ThemesProcessor->process()  #38 --/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(62): Grav\Framework\RequestHandler\RequestHandler->handle()  
#39 --/public_html/system/src/Grav/Common/Processors/PluginsProcessor.php(39): Grav\Framework\RequestHandler\RequestHandler->handle()  
#40 --/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(50): Grav\Common\Processors\PluginsProcessor->process()  #41 --/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(62): Grav\Framework\RequestHandler\RequestHandler->handle()  
#42 --/public_html/system/src/Grav/Common/Processors/InitializeProcessor.php(130): Grav\Framework\RequestHandler\RequestHandler->handle()  
#43 --/public_html/system/src/Grav/Common/Debugger.php(546): Grav\Common\Processors\InitializeProcessor::Grav\Common\Processors\{closure}()  #44 --/public_html/system/src/Grav/Common/Processors/InitializeProcessor.php(131): Grav\Common\Debugger->profile()  
#45 --/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(50): Grav\Common\Processors\InitializeProcessor->process()  
#46 --/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(62): Grav\Framework\RequestHandler\RequestHandler->handle()  
#47 --/public_html/system/src/Grav/Framework/RequestHandler/Middlewares/MultipartRequestSupport.php(40): Grav\Framework\RequestHandler\RequestHandler->handle()  
#48 --/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(50): Grav\Framework\RequestHandler\Middlewares\MultipartRequestSupport->process()  
#49 --/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(62): Grav\Framework\RequestHandler\RequestHandler->handle()  
#50 --/public_html/system/src/Grav/Common/Grav.php(312): Grav\Framework\RequestHandler\RequestHandler->handle() 
#51 --/public_html/index.php(47): Grav\Common\Grav->process()  
#52 {main} [] []

```

**Comment by rhukster** [comment]
Fixed with Email 4.0.0-rc.4
